File: Patterns/10. Hourglass Patterns/92. Hollow Hourglass/hh.py
n = int(input("Enter a Number : "))

for i in range(1,2*n):
  
    row = min(i,2*n-i)

    for j in range(1,2*n):
        if(row==j or row==1 or row+j==2*n):
            print("*",end=" ")
        else:
            print(end='  ')
    print()


'''
Output:

* * * * * * * * * 
  *           *   
    *       *     
      *   *       
        *         
      *   *       
    *       *     
  *           *   
* * * * * * * * *          

                
'''

File: Patterns/10. Hourglass Patterns/92. Hollow Hourglass/hh.java
import java.util.*;
public class hh{
    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Number : ");
        int n = sc.nextInt();

        for(int i=1; i<=2*n-1; i++){

            int row = Math.min(i,2*n-i);

            for(int j=1; j<=2*n-1; j++){
                if(row==j || row==1 || row+j==2*n){
                    System.out.print("* ");
                }else{
                    System.out.print("  ");
                }
            }
            System.out.println();
        }
    }
}

/*
Output

* * * * * * * * * 
  *           *   
    *       *     
      *   *       
        *         
      *   *       
    *       *     
  *           *   
* * * * * * * * *   

*/